service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -58,11 +62,46 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		}
 	}
 
+	var irsa bool
+	var accountID string
+	{
+		if r.provider == "aws" {
+
+			awsCluster := &v1alpha3.AWSCluster{}
+			err := r.ctrlClient.Get(ctx, types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, awsCluster)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			if _, ok := awsCluster.Annotations[k8smetadata.AWSIRSA]; ok {
+				irsa = true
+			}
+
+			secret, err := r.k8sClient.CoreV1().Secrets(awsCluster.Spec.Provider.CredentialSecret.Namespace).Get(ctx, awsCluster.Spec.Provider.CredentialSecret.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				r.logger.Debugf(ctx, "secret '%s/%s' not found cannot set accountID", cr.Namespace, key.APISecretName(&cr))
```

Comment:
shouldn't we return here if the secret is not found?

Reply:
Added a return 👍🏻